src/test/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheckTest.java

```diff
@@ -98,13 +102,30 @@ public void testDefault() throws Exception {
     @Test
     public void testNotFoundKey() {
         final List<String> testStrings = new ArrayList<>(3);
+        Method getLineNumber = null;
+        try {
+            getLineNumber = UniquePropertiesCheck.class.getDeclaredMethod(
+                "getLineNumber", FileText.class, String.class);
+        }
+        catch (NoSuchMethodException ex) {
+            LOG.debug("NoSuchMethodException occurred.", ex);
```

Comment:
This is almost similar to a try-catch-fail but worse. If method can't be found, you are logging the exception and the JUnit will still appear to pass. No one will look at logs for a failure, they only see success and don't care.

If method can't be found throw the exception out of the method to the runner so it will fail and display failure to user. There should be no try-catch.